glnx-errors.h

```diff
@@ -49,16 +49,47 @@ glnx_throw (GError **error, const char *fmt, ...)
   return FALSE;
 }
 
-/* Like glnx_throw(), but yields a NULL pointer. */
+/* Like `glnx_throw ()`, but returns %NULL. */
 #define glnx_null_throw(error, args...) \
   ({glnx_throw (error, args); NULL;})
 
+/* Implementation detail of glnx_throw_prefix() */
+void glnx_real_set_prefix_error_va (GError     *error,
+                                    const char *format,
+                                    va_list     args) G_GNUC_PRINTF (2,0);
+
+/* Prepend to @error's message by `$prefix: ` where `$prefix` is computed via
+ * printf @fmt. Returns %FALSE so it can be used conveniently in a single
+ * statement:
+ *
+ * ```
+ *   if (!function_that_fails (s, error))
+ *     return glnx_throw_prefix (error, "while handling '%s'", s);
+ * ```
+ * */
+static inline gboolean G_GNUC_PRINTF (2,3)
+glnx_throw_prefix (GError **error, const char *fmt, ...)
```

Comment:
A bit of a bikeshed but...the error was already thrown.  Maybe `glnx_prefix_error()`?